neo/Network/P2P/Peer.cs

```diff
@@ -179,25 +183,42 @@ private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
             ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
             if (MaxConnections != -1 && ConnectedPeers.Count >= MaxConnections && !TrustedIpAddresses.Contains(remote.Address))
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                DisconnectWithAddresses(DisconnectReason.MaxConnectionReached);
                 return;
             }
 
             ConnectedAddresses.TryGetValue(remote.Address, out int count);
             if (count >= MaxConnectionsPerAddress)
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                DisconnectWithAddresses(DisconnectReason.MaxConnectionPerAddressReached);
+                return;
             }
             else
             {
                 ConnectedAddresses[remote.Address] = count + 1;
                 IActorRef connection = Context.ActorOf(ProtocolProps(Sender, remote, local), $"connection_{Guid.NewGuid()}");
                 Context.Watch(connection);
                 Sender.Tell(new Tcp.Register(connection));
-                ConnectedPeers.TryAdd(connection, remote);
+                ConnectedPeers.TryAdd(connection, new List<IPEndPoint> { remote });
             }
         }
 
+        private void DisconnectWithAddresses(DisconnectReason reason)
```

Comment:
I did not understand this very well, can you clarify?

Reply:
> It could be: "Adding disconnection reason and its use for ensuring better p2p performance"

done

Reply:
> I did not understand this very well, can you clarify?

Should I change the method name to `DisconnectAndAttachedWithAddresses` ？

Reply:
Fixed, I removed this function, replaced by `Disconnect`.